src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -48,9 +48,10 @@ public Uri AppConfigurationEndpoint
         {
             get
             {
-                if (_options.Endpoint != null)
+                if (_options.Endpoints != null)
                 {
-                    return _options.Endpoint;
+                    // Split each endpoint with a '.' to get the config store name, and return the endpoint which doesn't have '---' (that would be the primary endpoint).
+                    return _options.Endpoints.First(c => !c.Host.Split('.')[0].Contains(ConfigurationStoreConstants.ConfigStoreNameReplicaSeparator));

```

Comment:
Again, there is no guarantee the primary endpoint is in the list.

Reply:
If the Endpoints don't have a primary endpoint, do we just strip the `---xyz` from the available endpoints and return it?

Reply:
That depends on who is using it and how it is used.

Reply:
I think we discussed returning the first endpoint in the list (may or may not be primary).

Reply:
With the new naming scheme for the replicas. (i.e. {storeName}-{replicaName}.azconfig.io), it is not possible to discern primary endpoint vs. replica. So changed the implementation to return the first endpoint in the list.